### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by kelvin-cao**
- Add project README
- Update fabric client examples README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Interface
+
+When configured, the Switchtec devices advertise a special DMA endpoint which
+enables the DMA functionality. BAR0 of the endpoint exports the firmware and
+hardware registers for the driver to configure and control the DMA engine.
+
+The switchtec dma driver maintains two queues in host memory space, a submission
+queue (SQ) and a completion queue (CQ), to communicate with the hardware DMA
+engine. The Switchtec DMA driver adds Submission Elements to the SQ on behalf of
```
Comment (ggec62160) [inline README.md]: Submission Elements (SE)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+the DMA client when a DMA transfer is kicked off. The DMA engine adds Completion
+Elements to the CQ when a DMA transfer has finished.
```
Comment (ggec62160) [inline README.md]: 1) append CE after Completion Elements.
2) is the CE added by device/hardware instead of DMA engine?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+### DMA client interface
+
+In basic mode, the Switchtec DMA driver works as a DMA controller driver
+(provider) of the Linux DMAEngine framework. For more detail, refer to
+Documentation/driver-api/dmaengine/provider.rst in the Linux kernel source tree.
+
+## Fabric mode
+
+In fabric mode, the Switchtec DMA driver enables data movement from one host to
+another within a Switchtec fabric.
```
Comment (ggec62160) [inline README.md]: with in the same Switchtec fabric?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+This mode is automatically turned on with Switchtec PAX DMA endpoint devices
+when they're probed. With this mode on, the Switchtec DMA driver enables the
+following DMA transactions:
+
+- Memory copy from the local host to a remote host
+- Write Immediate to a remote host
+- Remote Host Interrupt to a remote host
```
Comment (ggec62160) [inline README.md]: Send a Remote Host Interrupt to remove host?

File: README.md (hunk 1 above)
Comment (kelvin-cao) [reply README.md]: Will update.

File: README.md (hunk 2 above)
Comment (kelvin-cao) [reply README.md]: For 1. Will update. 
For 2. DMA engine is the DMAC/hardware. I think DMA engine is a common way to refer to this specific hardware piece (DMAC).

File: README.md (hunk 3 above)
Comment (kelvin-cao) [reply README.md]: Will update.

File: README.md (hunk 4 above)
Comment (kelvin-cao) [reply README.md]: How about "Interrupt a remote host (Remote Host Interrupt, or RHI)"?

**Pull request merged**